src/command/archive/get/file.c

```diff
@@ -53,24 +89,13 @@ archiveGetFile(
                 StorageWrite *const destination = storageNewWriteP(
                     storage, walDestination, .noCreatePath = true, .noSyncFile = true, .noSyncPath = true, .noAtomic = true);
 
-                // If there is a cipher then add the decrypt filter
-                if (actual->cipherType != cipherTypeNone)
-                {
-                    ioFilterGroupAdd(
-                        ioWriteFilterGroup(storageWriteIo(destination)),
-                        cipherBlockNewP(cipherModeDecrypt, actual->cipherType, BUFSTR(actual->cipherPassArchive)));
-                    compressible = false;
-                }
+                compressible = buildArchiveGetPipeLine(ioWriteFilterGroup(storageWriteIo(destination)), actual);
 
-                // If file is compressed then add the decompression filter
-                CompressType compressType = compressTypeFromName(actual->file);
-
-                if (compressType != compressTypeNone)
+                if (walIsSegment(request) && cfgOptionTest(cfgOptFilter))
                 {
-                    ioFilterGroupAdd(ioWriteFilterGroup(storageWriteIo(destination)), decompressFilterP(compressType));
-                    compressible = false;
+                    ioFilterGroupAdd(ioWriteFilterGroup(storageWriteIo(destination)),
+                                     walFilterNew(pgVersion, cfgOptionStrId(cfgOptFork), pageSize, actual));
```

Comment:
why not use `walSegmentSize` from `pgControl` like `pageSize`?

Reply:
The block size can also be obtained from the control file.

Reply:
> As I understand it, PgControl stores the page size of heap tables, and the page size of the WAL file is not there, but they can be different.

I didn't say that the block size is already in the PgControl structure, but it is in the control file.